### ISSUE TITLE
Unifying an argument in the `ReleaseUiaProvider` method to `HWND`.

### ISSUE THREAD

**Issue by NikitaSemenovAkvelon** [dotnet/winforms#8217]
Replace the type of the `handle` parameter from `IntPtr` to `IHandle<HWND>` in the Control.ReleaseUiaProvider method and its overrides. 

        Suggestion - consider unifying argument in this method to `nint` in the next PR.

_Originally posted by @Tanya-Solyanik in https://github.com/dotnet/winforms/pull/7349#discussion_r1023472589_
 📓 - modified bug title and body per @elachlan suggestion

**Comment by elachlan** [comment]
I feel for handles it should be obvious that its a handle. `nint` doesn't provide that. Maybe we can use `IHandle`?

**Comment by NikitaSemenovAkvelon** [comment]
@Tanya-Solyanik what do you think about @elachlan suggestion?

**Comment by Tanya-Solyanik** [comment]
Good point, UiaReturnRawElementProvider already moved to `IHandle<HWND>`. https://github.com/dotnet/winforms/blob/f4f931118878cbbf2a90deef3440767b8f0109e3/src/System.Windows.Forms.Primitives/src/Interop/UiaCore/Interop.UiaReturnRawElementProvider.cs#L14

**Comment by JeremyKuhne** [comment]
Be careful about boxing with `IHandle<HWND>`. In general we don't want to take `IHandle<HWND>` as a parameter to avoid boxing `HWND` to `IHandle<HWND>`. Using `T` with a constraint of `IHandle<HWND>` gets around the boxing problem, which is why we do it that way for the various `PInvoke` overloads.

**Comment by elachlan** [comment]
Ahhhh, ok. Well maybe in this case we just pass HWND. At least its a handle type. The main suggestion I was making was to avoid nint when dealing with handles so its obvious its a handle and not just an integer.

**Comment by elachlan** [comment]
This is really strange. Since its a `Control` which implements `IHandle<HWND>`, shouldn't `ReleaseUiaProvider` not have the `handle` parameter and pass itself to `UiaReturnRawElementProvider`?

**Comment by elachlan** [comment]
I agree, the design can definitely be improved here to avoid exposing the `IntPtr` parameter.
